[Button Overhaul]

RYin-28 opened the pull request:
![Screenshot 2023-11-01 012720](https://github.com/Jax-Donuts/Jax-website/assets/138041702/7590cd37-af31-4960-b068-c97989192ef5)
- Removed global settings for buttons
- Added a new button to the components file
- Created a new general button
- Created a new header button
- Deleted round button component
- Created a new submit button to replace the round button -- Renamed all instances of a round button to the submit button
- Edited footer static and hover colors
- Edited cards on the home page for color consistency
- Added general button properties to the basic button for the menu tab in the header
- Made appropriate changes to all buttons found throughout the project to new custom buttons

## Issue Name
Button overhaul

## Describe your changes
Created 3 new buttons to be used throughout the project
1. General button
- Used in cases where a basic button is needed with our stylings
2. Submit button
- Used in cases where a button needs an onClick action (such as form submissions)
3. Header button
- Used exclusively for the header. Has animation stylings (underline upon hovering).

Attempted to get color consistency across icons, texts, backgrounds, etc.
(note: different libraries interpret hex codes differently so I tried my best to color match everything)

## Checklist before requesting a review

- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.
- [X] Do we need to implement analytics?

RYin-28 committed:
[Button Overhaul]
- Removed global settings for button
- Added new button to components file
- Created new general button
- Created new header button
- Deleted round button component
- Created new submit button to replace round button
-- Renamed all instances of round button to submit button
- Edited footer static and hover colors
- Edited cards on home page for color consistency
- Added general button properties to basic button for menu tab in the header
- Made appropriate changes to all buttons found throughout project to new custom buttons

drmicrochip97 commented on src/app/contact/page.tsx:
```diff
@@ -32,7 +31,7 @@ export default function ContactUs() {
             <Stack mih="100%" spacing="xl" justify="space-between" align="stretch">
               <Card bg="#FFF5F5" radius="xl" sx={() => ({ textAlign: 'center' })}>
-                <ClockHour4 color="#fa5252" size="10rem" />
+                <ClockHour4 color="#FA5252" size="10rem" />
```
I think we should make a constant enum called like so
```ts
enum MainColors {
  LightRed="#FA5252"
}
```

You might be able to shove it in `src/shared/constants.ts`. And then hopefully you can reference it in this file like 
```tsx
<ClockHour4 color={MainColors.LightRed} size="10rem" />
```

drmicrochip97 replied:
If that doesn't work let me know. I think I've had issues in the past with the constants file.

drmicrochip97 replied:
I also noticed this color "#EF4547" is used a lot too. Maybe you can add it to the MainColors enum too!

drmicrochip97 replied:
I also noticed this color "#F15859". Which also looks very similar. Hmmmm.

drmicrochip97 commented on src/components/button/header-button.styles.ts:
```diff
@@ -0,0 +1,31 @@
+    position: 'relative',
+
+    '&:before': {
+      content: "''",
```
What's the empty string encased in another empty string?

RYin-28 replied:
I don't understand it entirely but I found the code here:
https://stackoverflow.com/questions/69730677/underline-animation-on-hover-mui

drmicrochip97 replied:
Seems to be related to Material UI, which I'm fairly certain we're not using

drmicrochip97 commented on src/components/button/submit-button.tsx:
```diff
@@ -0,0 +1,21 @@
+
+  return (
+    <>
+      <Button type="submit" variant="filled" radius="xl" className={classes.button} onClick={() => onClick()}>
```
nit:

The onClick can be rewritten a little cleaner
```tsx
<Button type="submit" variant="filled" radius="xl" className={classes.button} onClick={onClick}>
```

drmicrochip97 reviewed (changes requested):
I think it looks good. What we need to do is eliminate the constant use of the color strings. We should be referencing one source of truth for our colors that we choose. Especially since all the reds are kinda similar. Note the comment I made about the enum. 

Feel free to reach out if you encounter issues with the enum approach.

RYin-28 committed:
[Pull Request Changes]
- Created constant enums to simplify color implementation
- Replaced all hex codes with new constants
- Reduced number of color options

RYin-28 committed:
[Pull Request Changes]
- Created constant enums to simplify color implementation
- Replaced all hex codes with new constants
- Reduced number of color options

drmicrochip97 commented on src/app/admin/products-table/products-table.tsx:
```diff
@@ -64,12 +64,12 @@ export function ProductsTable({ products, openEdit, getProducts }: Props) {
         }
       >
-        <div style={{ border: '2px solid #fa5252', padding: 10, borderRadius: 5 }}>
+        <div style={{ border: '2px solid #FA5252', padding: 10, borderRadius: 5 }}>
```
If you need to do string formatting you can do.
```ts
`2px solid ${MainColors.RedStatic}`
```

 Notice the backticks instead of quotations.

drmicrochip97 commented on src/shared/constants.ts:
```diff
@@ -1,3 +1,11 @@
+  RedStatic = '#FA5252', // icons, delete form border, header, submit, general button color
+  RedHover = '#EF4547', // general, submit button hover color
+
+  // Search and replace theme.color
```
Is the comment still relevant?

drmicrochip97 reviewed (approved):
looks gucci. just a few minor things here.

RYin-28 committed:
Merge branch 'button-overhaul' of github.com:Jax-Donuts/Jax-website into button-overhaul
